doc/release-versioning.md

```diff
@@ -0,0 +1,151 @@
+    tags/"v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.0"
+$ git push git@github.com:operator-framework/operator-sdk.git \
+    "v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.x"
+```
+
+## Patch versions
+
+Patch versions changes are meant only for bug fixes, and will not break compatibility of the current minor version. A patch release will contain a collection of minor bug fixes, or individual major and security bug fixes, depending on severity.
+
+### Creating a patch version branch
+
+As more than one patch may be created per minor release, patch branch names of the form "${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}" will be created after a bug fix has been pushed, and the bug fix branch merged into the patch branch only after testing.
+
+```bash
+$ git checkout "v${MAJOR_VERSION}.${MINOR_VERSION}.x"
+$ git checkout -b "cherry-picked-change"
+$ git cherry-pick "$GIT_COMMIT_HASH"
+$ git push origin "cherry-picked-change"
+```
+
+# Releases
+
+Making an Operator SDK release involves:
+- Tagging and signing a git commit and pushing the tag to GitHub.
+- Building a release binary, signing the binary, and uploading both binary and signature to GitHub.
+
+Releases can only be performed by [maintainers][doc-maintainers].
+
+## Binaries
+
+Release binaries will be built for the `x86_64` architecture. Binaries will be signed using a maintainers' [PGP key][doc-maintainer-pgp-keys], and signatures will be uploaded to the release along with its accompanying tarball-ed binary. Ensure you import maintainer keys to verify release binaries. Note: `gpg` and `gpg2` can be used interchangeably to sign and verify binaries; substitute `gpg2` for `gpg` below.
+
+Creating a release binary tarball:
+```bash
+$ make build
+```
+
+Creating a release tarball:
+```bash
+$ tar \
+  -zcvf "operator-sdk.${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}.${TARGET_ARCH}.tar.gz" \
+  -C ./build \
+  .
+```
+
+Release signing:
+```bash
+$ gpg \
+  --output "operator-sdk.${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}.${ARCHITECTURE}.asc" \
+  --detach-sig "$OPERATOR_SDK_BINARY"
+```
+
+Release verification:
+```bash
+$ gpg \
+  --verify "operator-sdk.${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}.${ARCHITECTURE}.asc" \
+  "$OPERATOR_SDK_BINARY"
+```
+
+## Release tags
+
+Every release will have a corresponding git tag.
+
+Make sure you've [uploaded your GPG key][link-github-gpg-key-upload] and configured git to [use that signing key][link-git-config-gpg-key] either globally or for the Operator SDK repository. Note: the email the key is issued for must be the email you use for git.
+
+```bash
+$ git config [--global] user.signingkey "$GPG_KEY_ID"
+$ git config [--global] user.email "$GPG_EMAIL"
+```
+
+Create a signed tag for the commit you wish to release. If prompted to enter a tag message, enter the release version.
+
+```bash
+$ git tag \
+    -s "v${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}" \
+    "$GIT_COMMIT_HASH"
```

Comment:
There's a lot of different steps to perform to do a release and that leaves a lot of room for error. A script might make sense to help simplify this. As I mentioned in an above comment, ct has a pretty nice release process. They have a script that does various checks, creates and verifies a signed tag, and then runs `make release`. I feel that this would work well for our release process as well: https://github.com/coreos/container-linux-config-transpiler/blob/master/release.sh

Reply:
I fully agree. I'll take a look at that script and make adaptions.